From 9af9b0154e5c6dea14356c31ace112e6aaaa82e0 Mon Sep 17 00:00:00 2001
From: Khem Raj <khem_raj@cable.comcast.com>
Date: Wed, 8 Oct 2014 08:55:55 -0700
Subject: [PATCH 3/8] qtbase-native: Remove a race where icu-native was needed

We do not have ICU native in depchain so disable it
for native builds

Change-Id: I92283aaf5db3a9c29055dffa662b5dcd6ef09850
Signed-off-by: Khem Raj <khem_raj@cable.comcast.com>
---
 .../qt5/qtbase-5.1.1/0001-Always-build-uic.patch   | 40 ----------------------
 recipes-qt/qt5/qtbase-native-5.1.1.inc             |  3 +-
 2 files changed, 2 insertions(+), 41 deletions(-)
 delete mode 100644 recipes-qt/qt5/qtbase-5.1.1/0001-Always-build-uic.patch

diff --git a/recipes-qt/qt5/qtbase-5.1.1/0001-Always-build-uic.patch b/recipes-qt/qt5/qtbase-5.1.1/0001-Always-build-uic.patch
deleted file mode 100644
index 96c85f0..0000000
--- a/recipes-qt/qt5/qtbase-5.1.1/0001-Always-build-uic.patch
+++ /dev/null
@@ -1,40 +0,0 @@
-From f794542b10ccbf9c4cecaf23cf24658d64594cf3 Mon Sep 17 00:00:00 2001
-From: Mikko Levonmaa <mikko.levonmaa@palm.com>
-Date: Tue, 27 Nov 2012 12:46:44 -0800
-Subject: [PATCH 22/22] Always build uic
-
-Even if we are not building gui or widgets. This tool is needed later
-as a native tool when compiling the target.
-
-Change-Id: I257668ac28c22b192e7ec7736e6c23fa3be6bab6
-Signed-off-by: Mikko Levonmaa <mikko.levonmaa@palm.com>
-Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
----
- src/src.pro | 4 ++--
- 1 file changed, 2 insertions(+), 2 deletions(-)
-
-diff --git a/src/src.pro b/src/src.pro
-index b4c9602..f30ec1f 100644
---- a/src/src.pro
-+++ b/src/src.pro
-@@ -122,7 +122,7 @@ contains(QT_CONFIG, concurrent):SUBDIRS += src_concurrent
-     contains(QT_CONFIG, opengl(es1|es2)?):SUBDIRS += src_openglextensions
-     src_plugins.depends += src_gui src_platformsupport
-     !contains(QT_CONFIG, no-widgets) {
--        SUBDIRS += src_tools_uic src_widgets
-+        SUBDIRS += src_widgets
-         src_plugins.depends += src_widgets
-         contains(QT_CONFIG, opengl(es1|es2)?) {
-             SUBDIRS += src_opengl
-@@ -134,7 +134,7 @@ contains(QT_CONFIG, concurrent):SUBDIRS += src_concurrent
-         }
-     }
- }
--SUBDIRS += src_plugins src_tools_qdoc
-+SUBDIRS += src_plugins src_tools_qdoc src_tools_uic
- 
- nacl: SUBDIRS -= src_network src_testlib
- 
--- 
-1.8.3.1
-
diff --git a/recipes-qt/qt5/qtbase-native-5.1.1.inc b/recipes-qt/qt5/qtbase-native-5.1.1.inc
index 1adf422..472bbb5 100644
--- a/recipes-qt/qt5/qtbase-native-5.1.1.inc
+++ b/recipes-qt/qt5/qtbase-native-5.1.1.inc
@@ -28,7 +28,7 @@ SRC_URI += " \
 "
 
 # specific for native version
-SRC_URI += "file://0001-Always-build-uic.patch"
+SRC_URI += "file://0010-Always-build-uic.patch"
 
 QT_CONF_PATH = "${B}/qt.conf"
 
@@ -55,6 +55,7 @@ EXTRA_OECONF = " \
     -no-opengl \
     -no-openssl \
     -no-xcb \
+    -no-icu \
     -verbose \
     -release \
     -prefix ${OE_QMAKE_PATH_PREFIX} \
-- 
2.4.0

